<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1600</width>
    <height>1200</height>
   </rect>
  </property>
  <property name="baseSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="GLWidget" name="GLwidget" native="true">
      <property name="minimumSize">
       <size>
        <width>100</width>
        <height>100</height>
       </size>
      </property>
      <zorder>widget</zorder>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="widget" native="true">
      <property name="minimumSize">
       <size>
        <width>310</width>
        <height>310</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>190</width>
        <height>16777215</height>
       </size>
      </property>
      <widget class="QWidget" name="verticalLayoutWidget">
       <property name="geometry">
        <rect>
         <x>9</x>
         <y>10</y>
         <width>301</width>
         <height>1051</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <widget class="QGroupBox" name="algorithmsBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>60</height>
           </size>
          </property>
          <property name="sizeIncrement">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="baseSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="title">
           <string>Algorithms</string>
          </property>
          <widget class="QPushButton" name="relaxEmbeddingButton">
           <property name="geometry">
            <rect>
             <x>20</x>
             <y>40</y>
             <width>251</width>
             <height>41</height>
            </rect>
           </property>
           <property name="text">
            <string>Relax Embedding</string>
           </property>
          </widget>
          <widget class="QPushButton" name="pullButton">
           <property name="geometry">
            <rect>
             <x>90</x>
             <y>80</y>
             <width>111</width>
             <height>41</height>
            </rect>
           </property>
           <property name="text">
            <string>Pull</string>
           </property>
          </widget>
          <widget class="QGroupBox" name="renderingBox">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>150</y>
             <width>299</width>
             <height>138</height>
            </rect>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>40</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>16777215</height>
            </size>
           </property>
           <property name="title">
            <string>Rendering</string>
           </property>
           <widget class="QCheckBox" name="smoothShadeCheckBox">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>40</y>
              <width>301</width>
              <height>22</height>
             </rect>
            </property>
            <property name="text">
             <string>Smooth Shade</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
           <widget class="QCheckBox" name="wireframeCheckBox">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>70</y>
              <width>271</width>
              <height>22</height>
             </rect>
            </property>
            <property name="text">
             <string>Show Wireframe</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
           <widget class="QLabel" name="outputLabel">
            <property name="geometry">
             <rect>
              <x>10</x>
              <y>100</y>
              <width>201</width>
              <height>31</height>
             </rect>
            </property>
            <property name="text">
             <string>Output folder:</string>
            </property>
           </widget>
           <widget class="QLineEdit" name="outputEdit">
            <property name="geometry">
             <rect>
              <x>200</x>
              <y>100</y>
              <width>91</width>
              <height>31</height>
             </rect>
            </property>
           </widget>
          </widget>
          <widget class="QGroupBox" name="parameterBox">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>300</y>
             <width>299</width>
             <height>741</height>
            </rect>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>500</height>
            </size>
           </property>
           <property name="title">
            <string>Parameters</string>
           </property>
           <widget class="QGroupBox" name="physicalBox">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>50</y>
              <width>321</width>
              <height>231</height>
             </rect>
            </property>
            <property name="autoFillBackground">
             <bool>false</bool>
            </property>
            <property name="title">
             <string>Physical</string>
            </property>
            <property name="flat">
             <bool>false</bool>
            </property>
            <widget class="QLabel" name="youngsModulusLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>80</y>
               <width>281</width>
               <height>31</height>
              </rect>
             </property>
             <property name="text">
              <string>Young's Modulus:</string>
             </property>
            </widget>
            <widget class="QLabel" name="poissonRatioLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>110</y>
               <width>211</width>
               <height>41</height>
              </rect>
             </property>
             <property name="text">
              <string>Poisson Ratio:</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="poissonRatioEdit">
             <property name="geometry">
              <rect>
               <x>180</x>
               <y>120</y>
               <width>101</width>
               <height>31</height>
              </rect>
             </property>
            </widget>
            <widget class="QLabel" name="thicknessLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>150</y>
               <width>141</width>
               <height>51</height>
              </rect>
             </property>
             <property name="text">
              <string>Thickness:</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="thicknessEdit">
             <property name="geometry">
              <rect>
               <x>130</x>
               <y>160</y>
               <width>91</width>
               <height>31</height>
              </rect>
             </property>
            </widget>
            <widget class="QLabel" name="densityLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>200</y>
               <width>111</width>
               <height>31</height>
              </rect>
             </property>
             <property name="text">
              <string>Density:</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="densityEdit">
             <property name="geometry">
              <rect>
               <x>110</x>
               <y>200</y>
               <width>91</width>
               <height>31</height>
              </rect>
             </property>
            </widget>
            <widget class="QLabel" name="scaleLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>40</y>
               <width>171</width>
               <height>41</height>
              </rect>
             </property>
             <property name="text">
              <string>Scale:</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="youngsModulusEdit">
             <property name="geometry">
              <rect>
               <x>230</x>
               <y>80</y>
               <width>81</width>
               <height>31</height>
              </rect>
             </property>
            </widget>
            <widget class="QLineEdit" name="scaleEdit">
             <property name="geometry">
              <rect>
               <x>80</x>
               <y>50</y>
               <width>91</width>
               <height>31</height>
              </rect>
             </property>
            </widget>
            <zorder>youngsModulusLabel</zorder>
            <zorder>poissonRatioLabel</zorder>
            <zorder>poissonRatioEdit</zorder>
            <zorder>thicknessLabel</zorder>
            <zorder>thicknessEdit</zorder>
            <zorder>densityLabel</zorder>
            <zorder>densityEdit</zorder>
            <zorder>scaleLabel</zorder>
            <zorder>algorithmsBox</zorder>
            <zorder>solverBox</zorder>
            <zorder>youngsModulusEdit</zorder>
            <zorder>scaleEdit</zorder>
           </widget>
           <widget class="QGroupBox" name="solverBox">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>290</y>
              <width>301</width>
              <height>161</height>
             </rect>
            </property>
            <property name="title">
             <string>Solver</string>
            </property>
            <widget class="QLabel" name="eulerItersLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>50</y>
               <width>261</width>
               <height>31</height>
              </rect>
             </property>
             <property name="text">
              <string>Euler Iterations</string>
             </property>
            </widget>
            <widget class="QLabel" name="eulerTimestepLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>90</y>
               <width>251</width>
               <height>31</height>
              </rect>
             </property>
             <property name="text">
              <string>Euler Timestep</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="eulerItersEdit">
             <property name="geometry">
              <rect>
               <x>200</x>
               <y>50</y>
               <width>81</width>
               <height>31</height>
              </rect>
             </property>
            </widget>
            <widget class="QLineEdit" name="eulerTimestepEdit">
             <property name="geometry">
              <rect>
               <x>190</x>
               <y>89</y>
               <width>71</width>
               <height>31</height>
              </rect>
             </property>
            </widget>
            <widget class="QLabel" name="dampingCoeffLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>120</y>
               <width>221</width>
               <height>31</height>
              </rect>
             </property>
             <property name="text">
              <string>Damping Coeff</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="dampingCoeffEdit">
             <property name="geometry">
              <rect>
               <x>190</x>
               <y>120</y>
               <width>71</width>
               <height>31</height>
              </rect>
             </property>
            </widget>
           </widget>
           <widget class="QGroupBox" name="problemBox">
            <property name="geometry">
             <rect>
              <x>0</x>
              <y>460</y>
              <width>291</width>
              <height>191</height>
             </rect>
            </property>
            <property name="title">
             <string>Problem</string>
            </property>
            <widget class="QLabel" name="growthAmountLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>60</y>
               <width>211</width>
               <height>41</height>
              </rect>
             </property>
             <property name="text">
              <string>Growth Amount</string>
             </property>
            </widget>
            <widget class="QLabel" name="baseProbabilityLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>150</y>
               <width>221</width>
               <height>31</height>
              </rect>
             </property>
             <property name="text">
              <string>Base Probability</string>
             </property>
            </widget>
            <widget class="QLabel" name="maxStrainLabel">
             <property name="geometry">
              <rect>
               <x>0</x>
               <y>110</y>
               <width>151</width>
               <height>31</height>
              </rect>
             </property>
             <property name="text">
              <string>Max Strain</string>
             </property>
            </widget>
            <widget class="QLineEdit" name="growthAmountEdit">
             <property name="geometry">
              <rect>
               <x>210</x>
               <y>70</y>
               <width>71</width>
               <height>31</height>
              </rect>
             </property>
            </widget>
            <widget class="QLineEdit" name="maxStrainEdit">
             <property name="geometry">
              <rect>
               <x>140</x>
               <y>110</y>
               <width>91</width>
               <height>31</height>
              </rect>
             </property>
            </widget>
            <widget class="QLineEdit" name="baseProbabilityEdit">
             <property name="geometry">
              <rect>
               <x>210</x>
               <y>150</y>
               <width>81</width>
               <height>31</height>
              </rect>
             </property>
            </widget>
           </widget>
           <zorder>physicalBox</zorder>
           <zorder>solverBox</zorder>
           <zorder>algorithmsBox</zorder>
           <zorder>renderingBox</zorder>
           <zorder>problemBox</zorder>
          </widget>
          <widget class="QLabel" name="pullMagLabel">
           <property name="geometry">
            <rect>
             <x>10</x>
             <y>120</y>
             <width>191</width>
             <height>31</height>
            </rect>
           </property>
           <property name="text">
            <string>Pull Manitude:</string>
           </property>
          </widget>
          <widget class="QLineEdit" name="pullMagEdit">
           <property name="geometry">
            <rect>
             <x>200</x>
             <y>120</y>
             <width>101</width>
             <height>41</height>
            </rect>
           </property>
          </widget>
          <zorder>relaxEmbeddingButton</zorder>
          <zorder>pullButton</zorder>
          <zorder>renderingBox</zorder>
          <zorder>renderingBox</zorder>
          <zorder>parameterBox</zorder>
          <zorder>parameterBox</zorder>
          <zorder>pullMagLabel</zorder>
          <zorder>pullMagEdit</zorder>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1600</width>
     <height>46</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuVViewer">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionImport_OBJ"/>
    <addaction name="actionImport_Metric"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuView">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="actionReset_Camera"/>
    <addaction name="actionTake_Screenshot"/>
    <addaction name="actionExport_OBJ"/>
   </widget>
   <widget class="QMenu" name="menuActions">
    <property name="title">
     <string>Actions</string>
    </property>
    <addaction name="actionAdd_Noise"/>
    <addaction name="actionMinimize_with_Newton"/>
    <addaction name="actionSymmetrize"/>
    <addaction name="actionEigenvalues"/>
    <addaction name="separator"/>
    <addaction name="actionSet_No_Target_Metric"/>
    <addaction name="actionSet_Negative_K_Target_Metric"/>
    <addaction name="actionMake_Cone"/>
    <addaction name="actionMake_Flat_Cone"/>
    <addaction name="actionSet_Current_Lengths_as_Intrinsic"/>
   </widget>
   <addaction name="menuVViewer"/>
   <addaction name="menuView"/>
   <addaction name="menuActions"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionReset_Camera">
   <property name="text">
    <string>Reset Camera</string>
   </property>
  </action>
  <action name="actionTake_Screenshot">
   <property name="text">
    <string>Take Screenshot</string>
   </property>
  </action>
  <action name="actionExport_OBJ">
   <property name="text">
    <string>Export 3D Mesh</string>
   </property>
  </action>
  <action name="actionImport_OBJ">
   <property name="text">
    <string>Import OBJ</string>
   </property>
  </action>
  <action name="actionImport_Metric">
   <property name="text">
    <string>Import Metric</string>
   </property>
  </action>
  <action name="actionAdd_Noise">
   <property name="text">
    <string>Add Noise</string>
   </property>
  </action>
  <action name="actionSet_No_Target_Metric">
   <property name="text">
    <string>Set No Target Metric</string>
   </property>
  </action>
  <action name="actionSet_Negative_K_Target_Metric">
   <property name="text">
    <string>Set Negative K Target Metric</string>
   </property>
  </action>
  <action name="actionMinimize_with_Newton">
   <property name="text">
    <string>Minimize with Newton</string>
   </property>
  </action>
  <action name="actionSymmetrize">
   <property name="text">
    <string>Symmetrize</string>
   </property>
  </action>
  <action name="actionEigenvalues">
   <property name="text">
    <string>Eigenvalues</string>
   </property>
  </action>
  <action name="actionMake_Cone">
   <property name="text">
    <string>Make Cone</string>
   </property>
  </action>
  <action name="actionMake_Flat_Cone">
   <property name="text">
    <string>Make Flat Cone</string>
   </property>
  </action>
  <action name="actionSet_Current_Lengths_as_Intrinsic">
   <property name="text">
    <string>Set Current Lengths as Intrinsic</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
